handler/enqueue_annotation_test.go

```diff
@@ -0,0 +1,432 @@
+	var q workqueue.RateLimitingInterface
+	var instance EnqueueRequestForAnnotation
+	var mapper meta.RESTMapper
+	var pod *corev1.Pod
+	var podOwner = &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "podOwnerNs",
+			Name:      "podOwnerName",
+		},
+	}
+
+	BeforeEach(func() {
+		q = controllertest.Queue{Interface: workqueue.New()}
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "biz",
+				Name:      "biz",
+			},
+		}
+
+		err := SetOwnerAnnotation(podOwner, pod, schema.GroupKind{Group: "Pods", Kind: "core"})
+		Expect(err).To(BeNil())
+		Expect(cfg).NotTo(BeNil())
+		mapper, err = apiutil.NewDiscoveryRESTMapper(cfg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mapper).NotTo(BeNil())
+	})
+
+	Describe("EnqueueRequestForAnnotation", func() {
+		BeforeEach(func() {
+			instance = EnqueueRequestForAnnotation{
+				Type: schema.GroupKind{
+					Group: "Pods",
+					Kind:  "core",
+				}}
+		})
+
+		It("should enqueue a Request with the annotations of the object in the CreateEvent", func() {
+			evt := event.CreateEvent{
+				Object: pod,
+				Meta:   pod.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: podOwner.Namespace,
+					Name:      podOwner.Name,
+				},
+			}))
+
+		})
+
+		It("should enqueue a Request with the annotations of the object in the DeleteEvent", func() {
+			evt := event.DeleteEvent{
+				Object: pod,
+				Meta:   pod.GetObjectMeta(),
+			}
+			instance.Delete(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: podOwner.Namespace,
+					Name:      podOwner.Name,
+				},
+			}))
+		})
+
+		It("should enqueue a Request with annotations applied to both objects in UpdateEvent", func() {
+			newPod := pod.DeepCopy()
+			newPod.Name = pod.Name + "2"
+			newPod.Namespace = pod.Namespace + "2"
+
+			evt := event.UpdateEvent{
+				ObjectOld: pod,
+				MetaOld:   pod.GetObjectMeta(),
+				ObjectNew: newPod,
+				MetaNew:   newPod.GetObjectMeta(),
+			}
+
+			instance.Update(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: podOwner.Namespace,
+					Name:      podOwner.Name,
+				},
+			}))
+		})
+
+		It("should enqueue a Request with the annotations applied in one of the objects in UpdateEvent", func() {
+			newPod := pod.DeepCopy()
+			newPod.Name = pod.Name + "2"
+			newPod.Namespace = pod.Namespace + "2"
+			newPod.Annotations = map[string]string{}
+
+			evt := event.UpdateEvent{
+				ObjectOld: pod,
+				MetaOld:   pod.GetObjectMeta(),
+				ObjectNew: newPod,
+				MetaNew:   newPod.GetObjectMeta(),
+			}
+
+			instance.Update(evt, q)
+			Expect(q.Len()).To(Equal(1))
```

Comment:
EnqueueRequestForOwner enqueues 2 requests. We should probably do the same.

https://github.com/kubernetes-sigs/controller-runtime/blob/e1ced809069f5a88c7f7edd3b9fb67eda063ba39/pkg/handler/enqueue_owner.go#L67-L75

Reply:
Yes, we are doing the same [here](https://github.com/operator-framework/operator-lib/pull/2/files#diff-8b7f28521c3a813c9f60c46aba460128R97). I have added a [test](https://github.com/operator-framework/operator-lib/pull/2/files#diff-7afab47e92c4fe723633f2506d458bcaR130) to verify the same.